2020/02/21/consolidate-multiple-sftp-accounts-on-one-master-account.md

```diff
@@ -0,0 +1,154 @@
+---
+author: "Selvakumar Arumugam"
+title: "Consolidate Multiple SFTP Accounts on One Master Account"
+tags: ssh, sftp, shell
+---
+
+
+Recently, a data intensive client implemented a workflow process to generate various reports and insights from a list of facilities. Because a significant portion of these files contain sensitive data, they would need to strictly adhere to HIPAA compliance. Optimally, facilities should be able to transfer files securely and exclusively to our server. Arguably, the best method of achieving this is to create individual SSH File Transfer Protocol (SFTP) accounts for each source.
+
+
+### SFTP Account
+
+Private SFTP accounts were established for each facility and the data was received at a designated path. At these individual points of contact, a third party application picks up the data and processes further into the pipeline. The following demonstrates how SFTP accounts are developed and configured:
+
+
+* Create a user group for SFTP accounts
+
+```bash
+$ addgroup sftpusers
+```
+
+* Configure the following settings in sshd_config (this enables an SFTP account and defaults the location as the home path)
+
+```bash
+$ vi /etc/ssh/sshd_config
+
+Subsystem       sftp    internal-sftp
+
+Match Group sftpusers
+    ChrootDirectory /home/%u
+    AllowTCPForwarding no
+    X11Forwarding no
+    ForceCommand internal-sftp
+```
+
+* Restart SSH Server to apply changes
+
+```bash
+$ /etc/init.d/ssh restart
+```
+
+* Create an SFTP user account for a facility and place in a folder on the home path to receive data
+
+```bash
+useradd $USERNAME
+usermod -g sftpusers -s /usr/sbin/nologin $USERNAME
+mkdir -p /home/$USERNAME
+mkdir -p /home/$USERNAME/INPUT_PATH/
+chown -R root:root /home/$USERNAME
+```
+
+As the number of facilities increases, additional communication points are added. If the permissible amount of points is exceeded on the app by even one, it will not continue its service until more blocks are paid for. The implementation of this process will directly benefit our client.
```

Comment:
I don't understand what this is talking about. What are "permissible amount of points"? What payment is this referring to?

Reply:
As the number of facilities increases, additional communication points are created in third party application to interact with sftp accounts and pick up the files to process. At a point, The subscribed number of communication points limit is reached on the third party application and need to buy another block of communication points even to create one more. The implementation of following process directly benefited our client to comfortably stayed within existing subscribed limit.

Reply:
@jonjensen does this explanation makes sense ? 
FYI, third-party-app => Rhapsody

Reply:
@selvait90 It makes sense now, but I think the problem of billing in a third-party application may only be confusing here (at least it was for me). It doesn't seem important to what you're showing here, which is technically how to implement multiple SFTP accounts that feed into a master account. Do you think it would hurt anything to just remove that mention about number of facilities, and extra payment for a third-party app?

Reply:
@jonjensen 
I was trying to explain the reason for implementing this solution. But agreed with your point and remove that part.